### PR TITLE
feat: add routing and ability to navigate to a board

### LINKED ISSUE #37: As a user, I want to be able to view a particular Board
- [x] Set up react-router
- [x] Create route for `boards/:boardId`
- [x] Make each Board on the Boards page route properly on click.

### PR DESCRIPTION
**Pull request by benox3**
Resolves #37 

#### Tasks at a high level:
- add hash based routing
- link to a board page
- fetch board data
- routing decorator for storybook stories
- remove App placeholder story

### CODE REVIEW

File: src/components/Boards/components/BoardTile/BoardTile.styles.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { styled } from '@material-ui/core/styles';
```
Comment (benox3) [inline src/components/Boards/components/BoardTile/BoardTile.styles.tsx]: We can pull this out into shared styled components in the future.

Comment (remarkablemark) [reply src/components/Boards/components/BoardTile/BoardTile.styles.tsx]: What's the purpose of the muted link? Is this a stylistic choice? Let's make sure the app is still accessible

File: src/components/ErrorPage/ErrorPage.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import Box from '@material-ui/core/Box';
+import Typography from '@material-ui/core/Typography';
+
+export default function ErrorPage() {
+  return (
+    <Typography component="div">
+      <Box display="flex" height="100%" width="300" fontSize="h6.fontSize">
+        Oops... The link you are trying to reach can not be found
```
Comment (remarkablemark) [inline src/components/ErrorPage/ErrorPage.tsx]: `s/can not/cannot/`

File: src/components/ErrorPage/ErrorPage.tsx [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import Box from '@material-ui/core/Box';
+import Typography from '@material-ui/core/Typography';
+
+export default function ErrorPage() {
```
Comment (remarkablemark) [inline src/components/ErrorPage/ErrorPage.tsx]: Is error page solely for 404 error or does it encapsulate all kinds of errors?

File: src/components/Boards/components/BoardTile/BoardTile.styles.tsx (hunk 1 above)
Comment (benox3) [reply src/components/Boards/components/BoardTile/BoardTile.styles.tsx]: This just removes text decoration, shouldn't affect accessibility.

File: src/components/ErrorPage/ErrorPage.tsx (hunk 3 above)
Comment (benox3) [reply src/components/ErrorPage/ErrorPage.tsx]: Right now it's kind of a catch all, we can tweak once we have better handling in the future

File: src/components/Boards/components/BoardTile/BoardTile.styles.tsx (hunk 1 above)
Comment (remarkablemark) [reply src/components/Boards/components/BoardTile/BoardTile.styles.tsx]: 👍

File: src/components/ErrorPage/ErrorPage.tsx (hunk 3 above)
Comment (remarkablemark) [reply src/components/ErrorPage/ErrorPage.tsx]: Sounds good

**Comment by remarkablemark** [comment]
LGTM. It seems that CI failed the build because of coveralls but I think we can create a follow-up task to enforce 100% coverage.

**Comment by benox3** [comment]
> LGTM. It seems that CI failed the build because of coveralls but I think we can create a follow-up task to enforce 100% coverage.

Added and fixed up some tests. I have a follow up PR for adding storyshots which should add coverage for all the components render functions as well

**Pull request merged**